### PR TITLE
Added test cases for the coverage [MOSIP-25973]

### PR DESCRIPTION
**Pull request by GOKULRAJ136**
## Summary by CodeRabbit

* **Tests**
  * Added comprehensive unit tests for device specification logic and multiple provider implementations (modality mappings, version selection, discovery/availability, parsing, streaming/RCapture flows, and error paths).
  * Expanded software update test coverage for checksum, download/resume, rollback, manifest parsing, and upgrade flows.
  * Added REST client and response-signature tests including file-download/resume, signature validation, and connectivity checks.
  * Added tests for system properties, JSON/date mapping utilities, and various edge/error scenarios across registration services.

### CODE REVIEW

File: registration/registration-services/src/test/java/io/mosip/registration/test/mdm/service/MosipDeviceSpecification_SBI_1_0_ProviderImplTest.java [hunk 1]
```diff
@@ -197,96 +154,351 @@ public void streamTest() throws Exception{
...
+				.thenReturn("http://localhost/device");
+
+		when(mosipDeviceSpecificationHelper.getHttpClientResponseEntity(
+				anyString(), anyString(), anyString()))
+				.thenReturn("[]");
+
+		boolean result = mockObject.isDeviceAvailable(bioDevice);
+
+		assertFalse(result);
+	}
```
[the remark this answers is not in the document]
Comment (GOKULRAJ136) [reply registration/registration-services/src/test/java/io/mosip/registration/test/mdm/service/MosipDeviceSpecification_SBI_1_0_ProviderImplTest.java]: Noted. The test currently verifies the expected false result for an empty response. Adding an explicit objectMapper.readValue stub would improve clarity, but since the behavior is already covered and this is minor, we’ll retain it for now.

File: registration/registration-services/src/test/java/io/mosip/registration/test/mdm/service/MosipDeviceSpecificationFactoryTest.java [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+    public void getLatestVersion_singleNumber_returnsGreater() {
+        String result = invokePrivate(factory, "getLatestVersion", "2", "1");
+        assertEquals("2", result);
+    }
+
+    @Test
+    public void getLatestVersion_differentLength_returnsLonger() {
+        String result = invokePrivate(factory, "getLatestVersion", "1.0.0", "1.0");
+        assertEquals("1.0.0", result);
+    }
```
Comment (ase-101) [inline registration/registration-services/src/test/java/io/mosip/registration/test/mdm/service/MosipDeviceSpecificationFactoryTest.java]: Is this required? covering public methods itself should cover private methods @GOKULRAJ136

Comment (GOKULRAJ136) [reply registration/registration-services/src/test/java/io/mosip/registration/test/mdm/service/MosipDeviceSpecificationFactoryTest.java]: While private methods are usually tested indirectly, getLatestVersion() contains core comparison logic. Testing it directly helps validate all edge cases clearly.
